csharp/src/Microsoft.ML.OnnxRuntime/Training/CheckpointState.shared.cs

```diff
@@ -47,19 +35,129 @@ public CheckpointState(string checkpointPath)
         /// Loads Checkpoint state from path
         /// </summary>
         /// <param name="checkpointPath"> absolute path to checkpoint</param>
-        private void LoadCheckpoint(string checkpointPath)
+        public static CheckpointState LoadCheckpoint(string checkpointPath)
         {
-            NativeApiStatus.VerifySuccess(NativeTrainingMethods.OrtLoadCheckpoint(NativeOnnxValueHelper.GetPlatformSerializedString(checkpointPath), out handle));
+            if (!NativeTrainingMethods.TrainingEnabled())
+            {
+                throw new InvalidOperationException("Training is disabled in the current build. Please build ONNXRuntime from source with the build flag enable_training_apis.\n");
```

Comment:
Can change this message now to something like Install Microsoft.ML.OnnxRuntime.Training package

Reply:
Same applies to the error message in TrainingSession class